### PR TITLE
Seamless windows

### PR DESCRIPTION
**Pull request by Laboredih123**
## About the PR 
Makes windows seamless.
Adds a new proc for seaming and seamlessing (By pgmzeta)

## Why's this needed? 
It looks cool. People liked it.

## Images
image
image

## Changelog

```changelog
(u)Laboredih123 & pgmzeta
(+)Nanotrasen has now made windows fully connected, removing seams.
```

### CODE REVIEW

**Comment by ZeWaka** [comment]
This incorrectly affects standalone windows.

**Comment by ZeWaka** [comment]
Also definitely not a major changelog.

**Comment by Laboredih123** [comment]
After some testing, It doesn't seem to affect standalones for me.
image

**Comment by Laboredih123** [comment]
If you mean the single windows between walls, that is how the windows work.
image

**Comment by Eagletanker** [comment]
> If you mean the single windows between walls, that is how the windows work.
> image

The right window looks wronh

**Comment by Laboredih123** [comment]
I can assure you they are the same.

**Comment by Laboredih123** [comment]
image

**Comment by Laboredih123** [comment]
Although I notice a little seam.

**Comment by Laboredih123** [comment]
Should it be made that walls smooth with windows?

**Comment by ZeWaka** [comment]
> Should it be made that walls smooth with windows?

it either needs to smooth with the windows or the windows need to not smooth with the walls

**Comment by Laboredih123** [comment]
I find it nicer to smooth with the windows

**Comment by Laboredih123** [comment]
I'll ask.

**Comment by Laboredih123** [comment]
So it turns out windows are unable to connect to walls without smoothing.

**Comment by Laboredih123** [comment]
image
This looks quite nice, other than the fact it looks like the walls have a depression on top.

**Comment by pali6** [comment]
I prefer having the connector overlay on walls.

**Comment by Laboredih123** [comment]
People prefer the overlay over the without.

**Comment by ZeWaka** [comment]
Q: Why aren't we just applying the icon smoothing we use for walls, with the 40 states or whatever? Then we don't get window holes in 2x2 window cubes.

**Comment by pgmzeta** [comment]
> Q: Why aren't we just applying the icon smoothing we use for walls, with the 40 states or whatever? Then we don't get window holes in 2x2 window cubes.

Do we have a tool to generate the sprites for this? right now it's using icon_states in a dmi for each cardinal x ordinal combination:
image

If we have sprites for those, it's a *very* small lift to make sure we can use them with the refactoring changes in flight.

**Comment by Laboredih123** [comment]
I believe this would be a good addition to the pr. What do ya'll think? https://github.com/Laboredih123/goonstation/pull/1

**Comment by ZeWaka** [comment]
> > Q: Why aren't we just applying the icon smoothing we use for walls, with the 40 states or whatever? Then we don't get window holes in 2x2 window cubes.
> 
> Do we have a tool to generate the sprites for this?

Yes. https://github.com/tgstation/icon-cutter

File: code/obj/table.dm [hunk 1]
```diff
@@ -69,73 +69,42 @@
...
-		// west, south, and southwest
-		if (WT && ST)
-			var/obj/table/SWT = locate(src.auto_type) in get_step(src, SOUTHWEST)
-			if (SWT)
-				if (!src.working_image)
-					src.working_image = image(src.icon, "SW")
-				else
-					working_image.icon_state = "SW"
-				src.UpdateOverlays(working_image, "SWcorner")
+		if(16 & connections) // NE
```
Comment (ZeWaka) [inline code/obj/table.dm:77]: no, use the direction names.

**Review by ZeWaka — CHANGES_REQUESTED** [review]

**Pull request merged**